app/models/queen.rb

```diff
@@ -1,3 +1,15 @@
 class Queen < Piece
 
-end
+  def is_valid?(new_x, new_y)
+    return false if not super

```

Comment:
you can just call `super` here. it will automatically do what the super class does

Reply:
@ilyakrasnov I was trying that but it seemed that just calling super would not return what the parent method returns. For example, for the test of is_valid for a piece being off the board I would get true. Is there something I have to write in the parent method to avoid this?

Reply:
@eyoshimoriyama 100% correct, my bad 😄 . If you want a bit ruby sugar: `return false unless super`

Reply:
Oh nice I'll add that in!